### ISSUE TITLE
Cannot set an header to empty string

### ISSUE THREAD

**Issue by Steve-Mcl** [node-red/node-red#4789]
### Current Behavior

Setting (for example) `user-agent` header to an empty string is not possible.

### Expected Behavior

Apply provided value set in UI and `msg.headers`

### Steps To Reproduce

Add a HTTP-Req and try setting a header value to an empty string

### Environment

- Node-RED version: 3.x 4.x
- Node.js version:
- npm version:
- Platform/OS:
- Browser:

**Comment by Steve-Mcl** [comment]
Looking into this further, it is unclear if an empty header is valid or not.

The issue links to #4787 user-agent debate.

By not being able to set `user-agent` to an empty string, as of v4 it defaults to `Mozilla/5.0 (node-red)` which, according to the RFC is valid (and recommended) but seems to be somehow an issue for certain APIs Webservers.

I am now thinking rather than going down this particular rabbit hole right here and now we instead revert the default user-agent change which slipped in to PR #4616 (mea culpa, somehow 16 months ago when that branch was create it managed to slip through with the proxy changes)

**Comment by Steve-Mcl** [comment]
Parking this issue for now in light of the reverting the inclusion of a default user-agent string merged in #4791 

If/when it becomes a requirement, the change is simply to change https://github.com/node-red/node-red/blob/9d054543a765451a41b5959c1e5dba617097a93c/packages/node_modules/%40node-red/nodes/core/network/21-httprequest.js#L135 to 
```js
if(v || v === '') { headersObject[name] = v } //re-add with requested name & value
```
